test/unit/project_type/test_git.py

```diff
@@ -178,3 +178,15 @@ def expect_side_effect(patterns, *args, **kwargs):
         git_executor._execute_git_remote_command('fake command', cwd=None, timeout=0, log_msg_queue=MagicMock())
 
         self.assertTrue(matched_prompt, "The password prompt was not matched by pexpect")
+
+    def test_execute_raises_broken_pipe_error_on_last_try(self):
+        git_executor = _GitRemoteCommandExecutor()
+        self.patch('app.project_type.git.sleep')
+        mock_manager = self.patch('app.project_type.git.Manager')
+
+        def manager_init_side_effect(*args):
+            raise BrokenPipeError('')
```

Comment:
Does it work if you just do a direct assignment?

mock_manager.side_effect = BrokenPipeError('')